api/topn.md

```diff
@@ -14,18 +33,31 @@ do not use any experimental features in production.
 
 |Name|Type|Description|
 |-|-|-|
-|`agg`|`FrequencyAggregate`|The aggregate to display the values for|
-|`n`|`INTEGER`|The number of values to return|
-|`ty`|`AnyElement`|A value matching the type to output from the aggregate|
+|`agg`|`FrequencyAggregate` or `TopnAggregate`|The aggregate to display values for|
+|`n`|`INTEGER`|The number of values to return. Required only for frequency aggregates. For top N aggregates, defaults to target `n` of the aggregate itself|
+|`ty`|`AnyElement`|A value that provides an example of the output type|
+
+<highlight type="important">
+Because frequency aggregates and top N aggregates can operate over data of any
+type, `topn` uses the `ty` parameter to specify the data type of the output.
+`ty` must be an object, so it must be an actual example of the data type rather
+than the name of the type. For example, for integer output, use `0::INTEGER`.
+</highlight>
+
+## Optional arguments
+
+|Name|Type|Description|
+|-|-|-|
+|`n`|`INTEGER`|The number of values to return. Optional only for top N aggregates, where it must be less than the target `n` of the aggregate itself|
 
 ## Returns
 
 |Column|Type|Description|
 |-|-|-|
-|`topn`|`AnyElement`|The N most common elements seen in the aggregate.|
+|`topn`|`AnyElement`|The `n` most-frequent values in the aggregate.|
 
-If fewer than `N` elements have been found with the minimum
-frequency of the aggregate, this returns fewer than `N` values.
+If a frequency aggregate is used, and it contains fewer than `n` elements, the
+function returns fewer than `n` values.
```

Comment:
```suggestion
If there are fewer than `n` value available, either because there are not 
enough elements with the minimum frequency in a frequency aggregate, 
or the data isn't skewed enough to support a top n aggregate, the function 
returns fewer than `n` values.

Requesting more values from a top n aggregate than it was created for will
return an error.
```